src/library/dao/relationdao.cpp

```diff
@@ -0,0 +1,139 @@
+#include "library/dao/relationdao.h"
+
+#include <QString>
+
+#include "library/queryutil.h"
+#include "moc_relationdao.cpp"
+#include "track/relation.h"
+#include "util/db/fwdsqlquery.h"
+
+namespace {
+
+Relation* relationFromRow(const QSqlRecord& row) {
+    const auto id = DbId(row.value(row.indexOf("rl_id")));
+    TrackPair tracks = {
+            TrackId(row.value(row.indexOf("track_a"))),
+            TrackId(row.value(row.indexOf("track_b")))};
+    std::array<QVariant, 2> qPositions = {
+            row.value(row.indexOf("position_a")),
+            row.value(row.indexOf("position_b"))};
+    PositionPair positions = {
+            qPositions[0].isNull()
+                    ? std::nullopt
+                    : std::optional<mixxx::audio::FramePos>(qPositions[0].toInt()),
+            qPositions[1].isNull()
+                    ? std::nullopt
+                    : std::optional<mixxx::audio::FramePos>(qPositions[1].toInt())};
+    QString comment = row.value(row.indexOf("rl_comment")).toString();
+    QDateTime dateAdded = row.value(row.indexOf("rl_datetime_added")).toDateTime();
+
+    Relation* relation = new Relation(
```

Comment:
We try to avoid plain new. Maybe a std::unique_ptr here? Or a shared_ptr? 
Or return just by value. The compiler will likely optimize the copy away.